src/actions/security-chat.js

```diff
@@ -59,19 +87,67 @@ export function chatMessageIsTyping(boolean) {
     type: CHAT_MESSAGE_IS_TYPING,
   };
 }
-// export function chatRemoveMessage(messageId) {
-//   return {
-//     messageId: action.messageId,
-//     type: CHAT_REMOVE_MESSAGE,
-//   };
-// }
+export function chatRemoveMessage(messageId) {
+  return {
+    messageId,
+    type: CHAT_REMOVE_MESSAGE,
+  };
+}
+export function setUpdatedMessage(messageId, updatedMessage) {
+  return {
+    messageId,
+    message: updatedMessage,
+    isTyping: false,
+    type: CHAT_UPDATE_MESSAGE_BY_ID,
+  };
+}
+export function removeMessage(messageId) {
```

Comment:
This should look exactly the same way as `fetchMessages` or `updateMessage` below: first dispatch an action notifying that the request is started, then dispatch an action on success or an action on error. Here, instead of that, you'll not be notified in Redux of what's happening with a request.

Reply:
Ok. In examples in docs what you gave the result of the looks like: 
return await response.json();
With using redux I did:
export function removeMessage(messageId) {
  return async dispatch => {
    dispatch(fetchDataStart());
    try {
      const response = await fetch(
        `http://localhost:8080/api/message/${messageId}`,
        {
          method: 'delete',
        },
      );
      await response.json();
      return dispatch(chatRemoveMessage(messageId));
    } catch (error) {
      return dispatch(fetchDataFailure(error));
    }
  };
}
It`s right?

Reply:
And in reducer I need to return new state of messages array without deleted message. The best solution that I now is with "slice". Can you show me better solution please?

Reply:
Perfect, now make ALL functions you wrote in `actions` folder look like this.